File: android_code/ApiClient.java
package com.joange.cineapp.service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import com.joange.cineapp.api.PeliculaApiService;

public class ApiClient {
    
    // IMPORTANTE: Cambia esta URL por la IP de tu máquina
    // Para emulador Android: usa "http://10.0.2.2:8090/api/"
    // Para dispositivo físico: usa "http://TU_IP:8090/api/" (ejemplo: "http://192.168.1.100:8090/api/")
    private static final String BASE_URL = "http://10.0.2.2:8090/api/";
    
    private static Retrofit retrofit = null;
    
    public static Retrofit getRetrofit() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
    
    public static PeliculaApiService getApiService() {
        return getRetrofit().create(PeliculaApiService.class);
    }
}

File: app/src/main/java/com/example/buscaAnimeconexionspring/service/ApiClient.kt
package com.example.buscaAnimeconexionspring.service

import com.example.buscaAnimeconexionspring.api.AnimeApiService
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {

    // 👇 IMPORTANTE: puerto numérico + barra final
    // si tu backend va en 8091, cambia 8090 por 8091
    private const val BASE_URL = "http://10.0.2.2:8091/api/"

    private var retrofit: Retrofit? = null

    fun getRetrofit(): Retrofit {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)                         
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        }
        return retrofit!!
    }

    fun getAnimeApiService(): AnimeApiService {
        return getRetrofit().create(AnimeApiService::class.java)
    }
}
